### ISSUE TITLE
Class-scoped TypeVars should not be allowed in `self` annotation in `__init__` method

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7683]
The typing spec indicates that the type annotation for an `__init__` method should not include any class-scoped type variables. The following should generate an error:

```python
class Class8(Generic[T1, T2]):
    def __init__(self: "Class8[T2, T1]") -> None:  # E
        pass
```

**Comment by erictraut** [comment]
This is included in pyright 1.1.359, which I just published.
